### ISSUE TITLE
[Question] Sanitize version removing certain characters

### ISSUE THREAD

**Issue by earandap** [nebula-plugins/nebula-release-plugin#143]
I have a pipeline to build a Docker image in every push to specific branches in my git repository. The problem is that semantic versioning is using '+' character to specifying the metadata section in the version name. I need to replace this char for another supported by Docker in the image name. Is there any way to replace that character or use a custom version name?

```
jib {
    to {
        image = "registry.gitlab.com.uy:5005/project/app:$version"
        auth {
            username = System.getenv('CI_REGISTRY_USER')
            password = System.getenv('CI_REGISTRY_PASSWORD')
        }
        tags = ['latest']
    }
    container {
        ports = ['8080']
        environment = [
                SPRING_OUTPUT_ANSI_ENABLED: 'ALWAYS',
        ]
        useCurrentTimestamp = true
    }
    allowInsecureRegistries = true
}

jibDockerBuild.dependsOn bootJar
```
This is the error:

> * What went wrong:
> Execution failed for task ':jib'.
> > com.google.cloud.tools.jib.image.InvalidImageReferenceException: Invalid image reference: registry.gitlab.com.uy:5005/project/app:1.0.0-rc.1.dev.0+108db18

**Comment by EarthCitizen** [comment]
@earandap We are also having an issue with the `+` in the version name. In our case, it is causing problems when the JARs are opened as resource via URL.

**Comment by sdavids13** [comment]
Running into the exact same problem with have the `+` get uploaded to Maven and URL encoded. I would like to be able to get rid of any "special" character. I was trying to update the `version` on `afterEvaluate` of the release plugin initialization but using Kotlin script I could not access the `DelayedVersion` class (since it is Private) to be able to update the nested `ReleaseVersion inferredVersion` object. I just tried doing this which seems to be doing the trick:

```
release {
    afterEvaluate {
        val nonStandardCharacters = Regex("[^A-Za-z0-9_.-]")
        if(version.toString().contains(nonStandardCharacters)) {
            val newVersion =  version.toString().replace(Regex("[^A-Za-z0-9_.-]"), ".")
            rootProject.allprojects {
                version = newVersion
            }
            println("Pre-release version has changed to $newVersion to avoid problematic special characters for Maven/Docker/S3")
        }
    }
}
```

In an ideal world there would be some sort of configurable override that we can pass to plugin to avoid adding the `+` sign. If anyone has a better/cleaner approach to solving this problem I am all-ears.

**Comment by EarthCitizen** [comment]
@earandap Can you remove the reference to Docker in the subject of the issue so that the maintainers do not think that this question only relates to Docker? --Thanks

**Comment by rpalcolea** [comment]
Hi! sorry for the delay, we've been busy on this side.

We released `11.0.0` with a flag to sanitize the version. Please refer to https://github.com/nebula-plugins/nebula-release-plugin#sanitize-versions for its usage.
